packages/connect/src/base/entry.ts

```diff
@@ -380,58 +397,75 @@ export class EntryNodes extends EventEmitter {
     }
   }
 
-  /**
-   * Attempts to connect to a relay node
-   * @param id peerId of the node to dial
-   * @param relay multiaddr to perform the dial
-   * @param timeout when to timeout on unsuccessful dial attempts
-   * @returns a PeerStoreEntry containing the measured latency
-   */
-  private async connectToRelay(
-    id: PeerId,
-    relay: Multiaddr,
+  private async establishNewConnection(
+    destination: PeerId,
+    destinationAddress: Multiaddr,
     timeout: number
-  ): Promise<{ entry: EntryNodeData; conn?: Connection }> {
+  ): Promise<ConnResult | void> {
     const abort = new AbortController()
-    const start = Date.now()
-    let connectionEstablished = false
+    let done = false
 
     setTimeout(() => {
-      if (!connectionEstablished) {
+      if (!done) {
         abort.abort()
       }
     }, timeout)
 
     let conn: Connection | undefined
     try {
-      conn = await this.dialDirectly(relay, { signal: abort.signal })
+      conn = await this.dialDirectly(destinationAddress, { signal: abort.signal })
     } catch (err: any) {
-      error(`error while contacting entry node ${id.toB58String()}.`, err.message)
-      conn && (await attemptClose(conn, error))
-    } finally {
-      // Prevent timeout from doing anything
-      connectionEstablished = true
+      error(`error while contacting entry node ${destination.toB58String()}.`, err.message)
```

Comment:
In other places you moved to using `err` instead of `err.message`. What should we use now?

Reply:
Depends - if the source of the error is interesting for debugging purposes, the Error object gets logged. In all other cases, `err.message` is sufficient, such as here.